firmware/raspi/pieces.py

```diff
@@ -0,0 +1,228 @@
+        if self.canCastle:
+            i = 0
+            for x, y in possible_castles:
+                if not (rooks[i] or checks[i]):
+                    newLoc = beginningpos[0], beginningpos[1] + y
+                    betweenLoc = beginningpos[0], beginningpos[1] + (y / 2)
+                    if i == 0:
+                        knightLoc = beginningpos[0], beginningpos[1] + y - 1
+                        if board.alpha_notation(knightLoc) not in prohibited:
+                            if board.alpha_notation(newLoc) not in prohibited and board.alpha_notation(betweenLoc) not in prohibited:
+                                allowed_moves.append(newLoc)
+                    elif board.alpha_notation(newLoc) not in prohibited and board.alpha_notation(betweenLoc) not in prohibited:
+                        allowed_moves.append(newLoc)
+                i += 1
+
+        allowed_moves = filter(board.is_on_board, allowed_moves)
+        return map(board.alpha_notation, allowed_moves)
+
+
+class Queen(Piece):
+    """Queen chess piece.
+    """
+    shortname = "q"
+
+    def moves_available(self, pos):
+        """Returns moves available based on piece
+        """
+        return super(Queen, self).moves_available(pos.upper(), True, True, 8)
+
+class Rook(Piece):
+    """Rook chess piece.
+    """
+    shortname = "r"
+
+    def moves_available(self, pos):
+        """Returns moves available based on piece
+        """
+        return super(Rook, self).moves_available(pos.upper(), True, False, 8)
+
+class Bishop(Piece):
+    """Bishop chess piece.
+    """
+    shortname = "b"
+
+    def moves_available(self, pos):
+        """Returns moves available based on piece
+        """
+        return super(Bishop, self).moves_available(pos.upper(), False, True, 8)
+
+class Knight(Piece):
+    """Knight chess piece.
+    """
+    shortname = "n"
+
+    def moves_available(self, pos):
+        """Returns moves available based on piece
+        """
+        board = self.board
+        allowed_moves = []
+        beginningpos = board.num_notation(pos.upper())
+        piece = board.get(pos.upper())
+        deltas = (
+        (-2, -1), (-2, 1), (-1, -2), (-1, 2), (1, -2), (1, 2), (2, -1), (2, 1))
+        for x, y in deltas:
+            dest = beginningpos[0] + x, beginningpos[1] + y
+            if board.alpha_notation(dest) not in board.occupied(piece.color):
+                allowed_moves.append(dest)
+        allowed_moves = filter(board.is_on_board, allowed_moves)
+        return map(board.alpha_notation, allowed_moves)
+
+class Pawn(Piece):
+    """Pawn chess piece.
+    """
+    shortname = "p"
+
+    def moves_available(self, pos):
+        """Returns moves available based on piece
+        """
+        board = self.board
+        _ = self
+        if self.color == "white":
+            startpos, direction, enemy = 1, 1, "black"
+        else:
+            startpos, direction, enemy = 6, -1, "white"
+        allowed_moves = []
+        # Moving
```

Comment:
The function name is moves_available (which makes it seem like it would return the allowed_moves) but this comment makes it seem like this function is moving the pieces, is that what's happening?